ksp_plugin/planetarium.cpp

```diff
@@ -117,10 +119,113 @@ RP2Lines<Length, Camera> Planetarium::PlotMethod1(
     }
     new_rp2_lines.push_back(std::move(new_rp2_line));
   }
-  LOG(INFO) << "Skipped " << skipped << " points out of " << total;
+  LOG(INFO) << "PlotMethod1 skipped " << skipped << " points out of " << total
+            << ".";
   return new_rp2_lines;
 }
 
+RP2Lines<Length, Camera> Planetarium::PlotMethod2(
+    DiscreteTrajectory<Barycentric>::Iterator const& begin,
+    DiscreteTrajectory<Barycentric>::Iterator const& end,
+    Instant const& now) const {
+  RP2Lines<Length, Camera> lines;
+  if (begin == end) {
+    return lines;
+  }
+  auto last = end;
+  --last;
+  if (begin == last) {
+    return lines;
+  }
+
+  auto const plottable_spheres = ComputePlottableSpheres(now);
+
+  auto const final_time = last.time();
+  auto const& trajectory = *begin.trajectory();
+
+  double const tan²_angular_resolution =
+      Pow<2>(parameters_.tan_angular_resolution_);
+
+  auto previous_time = begin.time();
+  auto previous_degrees_of_freedom = begin.degrees_of_freedom();
+
+  Position<Navigation> previous_position;
+  Velocity<Navigation> previous_velocity;
+  Time Δt;
+
+  previous_position =
+      plotting_frame_->ToThisFrameAtTime(previous_time)
+          .rigid_transformation()(previous_degrees_of_freedom.position());
+  previous_velocity =
+      plotting_frame_->ToThisFrameAtTime(previous_time)
+          .orthogonal_map()(previous_degrees_of_freedom.velocity());
```

Comment:
Also let's decide on an order and use it consistently, e.g., time/position/velocity.  Compare lines 203-206.